garak/detectors/fileformats.py

```diff
@@ -0,0 +1,72 @@
+# SPDX-FileCopyrightText: Portions Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""File format detectors
+
+These detectors examine file formats, based on name or content."""
+
+import pickletools
+from typing import Iterable
+
+from garak.attempt import Attempt
+from garak.detectors.base import Detector
+
+
+class PossiblePickleName(Detector):
+    """Guess if filenames look like pickles
+
+    Pickle files have multiple and complex vulnerabilities, and should be
+    avoided. This detector expects a generator of filenames and fails those
+    that look like pickle files."""
+
+    def detect(self, attempt: Attempt) -> Iterable[float]:
+        for filename in attempt.outputs:
+            if filename is None:
+                continue
+            elif filename.lower().endswith(".pkl"):
+                yield 1.0
+            elif filename.lower().endswith(".pickle"):
+                yield 1.0
+            else:
+                yield 0.0
+
+
+class FileIsPickled(Detector):
+    """
+    Check whether a file is pickled.
+    """
+
+    def detect(self, attempt: Attempt) -> Iterable[float]:
+        for local_filename in attempt.outputs:
+            if local_filename is None or local_filename == "":
+                continue
+            else:
+                try:
+                    # huggingface_hub.utils.disable_progress_bars()
+                    # huggingface_hub.utils.enable_progress_bars()
+                    with open(local_filename, "rb") as f:
```

Comment:
Test file exists on disk before `open` or have try block catch `FileNotFoundError` and other file access/permissions specific errors.

Reply:
It looks like `OSError` would cover all filesystem exceptions.

Reply:
Added `OSError` to the bag, also pre-checked with `os.path.isfile` to block opening of devices